src/Lecture4.hs

```diff
@@ -135,7 +140,33 @@ errors. We will simply return an optional result here.
 -}
 
 parseRow :: String -> Maybe Row
-parseRow = error "TODO"
+parseRow = readRow . splitCSVLine
+   where readRow :: [String] -> Maybe Row
+         readRow cols
+            | length cols /= 3 = Nothing
+            | otherwise = Row <$> readProduct c1 <*> readTradeType c2 <*> readInt c3
+               where c1 = head cols
+                     c2 = cols !! 1
+                     c3 = last cols
+         readProduct :: String -> Maybe String
+         readProduct p
+            | dropSpaces p == "" = Nothing
+            | otherwise = Just p
+         readTradeType :: String -> Maybe TradeType
+         readTradeType trade = case dropSpaces trade of
+            "Sell" -> Just Sell
+            "Buy"  -> Just Buy
+            _      -> Nothing
```

Comment:
You don't need to trim spaces manually. And you can use `readMaybe` for the `TradeType` as well because the `readMaybe` function already does this for you 👾 

```haskell
λ: readMaybe "    42    " :: Maybe Int
Just 42
```